### PR TITLE
Adds ability to update todos

### PR DESCRIPTION
**Pull request by hassox**
Hi Mick,

I've added the ability to update the todos in the example app for sc resource tools. One thing I wasn't quite sure of is how check the status on updating (i.e. fetched, unfetched, saving etc) to incorporate these states of the record into the actions available.

Also there was a bug when creating records, where when the data came back from the server, the id attribute wasn't being updated. I manually update it in the done callback when creating, but I'm wondering if there's a built in way that this should happen with the resource?

Cheers
Daniel

### CODE REVIEW

File: public/js/app.js [hunk 1]
```diff
@@ -14,11 +14,17 @@ Todos.todosController = SC.ResourceCollection.create({
...
     var todo = Todos.Todo.create({ title: title || '', isDone: false });
-    todo.save().done(function() {
+    todo.save().done(function(data) {
+      todo.set('id', data.id);
       Todos.todosController.pushObject(todo);
     })
   },
 
+  updateTodo: function(todo){
+    if (!todo && todo.save) return;
```
Comment (staugaard) [inline public/js/app.js:24]: you mean `if (!todo || !todo.save) return;` or `if (!(todo && todo.save)) return;` right?

**Comment by staugaard** [comment]
Hey Daniel,

This example app was running on a very old version of SC.Resource. I just updated it, so now new object do get their id when saved. The way it works is that the response body from the save is applied to the instance. If no body is returned, SC.Resource will look for a Location header for the id.

What are you using SC.Resource for?

Thanks,
Mick

File: public/js/app.js (hunk 1 above)
Comment (hassox) [inline public/js/app.js:24]: Yes that's what I meant! It was pretty late ;)

**Comment by hassox** [comment]
Hi Mick,

I just round SC.Resource yesterday so I was having a look. We're building out a sproutcore app at work, and we started with the SC Datastore but we weren't really very happy with it so we pulled it out. We don't do too much crud though so it's working pretty well without a datastore. I've been looking at backbone and spine as well trying to get a handle on how the different frameworks are put together and how to use them. 

Really I'm just looking at the moment and it was great to try it out. It looks and feels pretty neat :) Thanks for writing it.

Are you using it in production at the moment?

Cheers
Daniel

**Comment by staugaard** [comment]
We are going into beta with a pretty big SC app very soon, and SC.Datastore was just way too big for us. So yeah we are almost in production with a big app on SC.Resource.

**Comment by scottmessinger** [comment]
The demo app on heroku has the old version of SC.Resource and the objects aren't updating with an ID when saved. As a result, it's impossible to delete tasks on the heroku app. 

Great wok, btw! I think the project looks super useful.

**Pull request closed without merging**